### PR TITLE
Article: Online-vs-Offline-Editor

### PR DESCRIPTION
**Pull request by ScrumUser1**

### CODE REVIEW

**Comment by alayek** [comment]
In future, don't open duplicate PRs like this, which is a duplicate of https://github.com/FreeCodeCamp/wiki/pull/642

Also, squash your commits. You have 9 commits, bundle them into a single one.

File: extra-3 [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (alayek) [inline extra-3]: What is this file for?

File: extras [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (alayek) [inline extras]: Ditto as above.

File: Online-vs-Offline-Editors [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+The only advantages of online IDE is that projects can be access everywhere; there are no setup and configurations. 
+There is no hardware limitation since all information will be stores in cloud.
+
+Links:
+Eclipse - https://www.app5.unisys.com/offerings/ClearPathConnection/eclipse.html
+Netbeans - https://netbeans.org/features/
+CodeRun - http://www.webappers.com/2009/12/04/coderun-studio-free-cross-platform-online-ide/
+IntelliJ Idea - https://www.jetbrains.com/idea/
+
```
Comment (alayek) [inline Online-vs-Offline-Editors]: Remove all the blank lines below.

File: Online-vs-Offline-Editors [hunk 4]
```diff
@@ -0,0 +1,71 @@
+**Online Offline Editors**
+**There are several offline IDE that programmers use in order to stay more productive to learn and be more productive at work**
```
Comment (alayek) [inline Online-vs-Offline-Editors]: Like I said in the last PR, this is a non-standard term. Define what you mean by these. I have no idea what offline IDE is, or what online IDE is.

File: Online-vs-Offline-Editors [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+team will need a reasonable office internet service. 
+
+If there is a major storm the connection will be affected causing many projects to be unfinished and
+incomplete. 
+
+The only advantages of online IDE is that projects can be access everywhere; there are no setup and configurations. 
+There is no hardware limitation since all information will be stores in cloud.
+
+Links:
+Eclipse - https://www.app5.unisys.com/offerings/ClearPathConnection/eclipse.html
```
Comment (alayek) [inline Online-vs-Offline-Editors]: Link like a pro.

> [Eclipse](https://www.app5.unisys.com/offerings/ClearPathConnection/eclipse.html)

File: Online-vs-Offline-Editors [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+Netbeans - https://netbeans.org/features/
+CodeRun - http://www.webappers.com/2009/12/04/coderun-studio-free-cross-platform-online-ide/
+IntelliJ Idea - https://www.jetbrains.com/idea/
```
Comment (alayek) [inline Online-vs-Offline-Editors]: When you are using Markdown, use its features instead of writing MS Word style documents in plain text.

**Comment by alayek** [comment]
@abhisekp @koustuvsinha if you get a chance, please do a review.

**Comment by koustuvsinha** [comment]
please change this file to a markdown file by adding a .md extension at the end.

@abhisekp @raisedadead i see a lot of PR's people just submit a plain null extension text file. why are people missing the fact that we use markdown files? I think it needs to be communicated again

**Comment by raisedadead** [comment]
@koustuvsinha 

> @abhisekp @raisedadead i see a lot of PR's people just submit a plain null extension text file. why are people missing the fact that we use markdown files? I think it needs to be communicated again

While I agree, lets take that discussion in the Chat, and keep this thread clean for this PR?

File: extra-3 (hunk 1 above)
Comment (ScrumUser1) [inline extra-3]: this was just an extra copy of the article i dont know how to remove it

Comment (alayek) [inline extra-3]: Just remove the file. As in, delete it, then commit your changes (deleting a file from your repo is a change), then squash commits properly.

**Comment by Rafase282** [comment]
I was working on making fixes but I noticed that there are many extra files being created and renamed and so on that it would be better to just create a fresh PR.

**Comment by Rafase282** [comment]
These are the changes I made before closing this bellow. @ScrumUser1 please contact me so you can use the material bellow and open a new PR. I'll help you.

---------------------------
# Online vs Offline IDE

## Offline IDEs

There are several offline IDE that programmers use in order to stay more productive to learn and be more productive at work.

- **[Eclipse](https://www.app5.unisys.com/offerings/ClearPathConnection/eclipse.html)** is more than a Java editor. Its biggest advantages are code completion by tabbing method which saves a lot of time writing documentations.

  It has a build in syntax check to fix any mistype words while coding a project. All of features we've come to expect of an IDE are their code completion, templates, integration  with various SCMSs, and integration with build systems.

  Its code formatting and cleanup tools are very powerful. I find its build system to work well and intuitively. I think these are the things upon which its reputation is really built.

  Furthermore it has a refactoring that find and replace function, variables and classes of a specific work.

  Eclipse offers more plugins, platforms, frameworks, and simpler to use compare  to other IDE such as **IntelliJ**, **BlueJ** and **JEdit**.

- **[NetBeans](https://netbeans.org/features/)** is a free intuitive editor that does it all compared from other IDE. It has simple Swing GUI design tool to build user interfaces by dragging and dropping components for instance buttons, textboxes.

  The _disadvantage_ it is that with all its features it is slow to load and uses  more memory than other IDE's.

- **[IntelliJ IDEA](https://www.jetbrains.com/idea/)** it is the most efficient offline editor however it is not free. It is faster than most editors however it has numbers of problems. Some major advantages is using too much memory from system,
- **[BlueJ](http://www.bluej.org/)** is a simple editor mostly used to teach java and object oriented programming. The main advantage of the editor over other ones is that it does not need a main method to run the program.

## Online IDEs

- **[CodeRun Studio](http://www.webappers.com/2009/12/04/coderun-studio-free-cross-platform-online-ide/)** is an online IDE that you build web application. It allows you to run and test code online.

  The disadvantage is that there is a 14 days trial then it must be paid. It is difficult to save your project since you must pay in order to fully operate it.

## Conclusion

There are several differences between online editors and the offline editors.

An offline IDE will be more beneficial to a programmer rather than an online IDE due to the fact that large projects will slow down, and a software team will need a reasonable office internet service.

If there is a major storm the connection will be affected causing many projects to be unfinished and incomplete.

The only advantages of online IDE is that projects can be access everywhere; there are no setup and configurations.  There is no hardware limitation since all information will be stores in cloud.

**Pull request closed without merging**